### PR TITLE
Acknowledge that we'll never be able to fake methods on generic types that have in parameters before .NET 6

### LINKED ISSUE #1382: Allow Faking of generic types with methods that have in parameters
Split from #1338.

As mentioned in #1337 and castleproject/Core#339, generic types containing a method with an `in` parameter cannot be faked by FakeItEasy. The ultimate cause is https://github.com/dotnet/corefx/issues/29254.

We should support this as soon as we can.

### PR DESCRIPTION
**Pull request by blairconrad**
Closes #1382.

### CODE REVIEW

File: tests/FakeItEasy.Specs/InParametersSpecs.cs [hunk 1]
```diff
@@ -79,9 +79,7 @@ public void SettingInParamDelegate(DelegateWithInParam fake, Exception exception
 
 #if LACKS_FAKEABLE_GENERIC_IN_PARAMETERS
```
Comment (blairconrad) [inline tests/FakeItEasy.Specs/InParametersSpecs.cs:80]: I thought about removing this section and just not testing, but I suppose there's a very small chance something will change. We might as well find out. But if you prefer, we could just to the `#else` part.

File: docs/what-can-be-faked.md [hunk 2]
```diff
@@ -43,4 +44,4 @@ Note that this means that static members, including extension methods,
 
 ### Methods that return values by reference
 
-Methods that return values by reference (officially called "[reference return values](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/ref-returns#what-is-a-reference-return-value)") cannot be invoked on a Fake. Any attempt to do so will result in a `NullReferenceException` being thrown.
\ No newline at end of file
+Methods that return values by reference (officially called "[reference return values](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/ref-returns#what-is-a-reference-return-value)") cannot be invoked on a Fake. Any attempt to do so will result in a `NullReferenceException` being thrown.
```
Comment (thomaslevesque) [inline docs/what-can-be-faked.md:47]: I don't understand why this line is shown as a difference. It's the same on both sides 🤔

**Review by thomaslevesque — APPROVED** [review]
> Acknowledge that we'll never be able to fake methods on generic types that have in parameters

You mean "before .NET 6", right?

File: docs/what-can-be-faked.md (hunk 2 above)
Comment (blairconrad) [reply docs/what-can-be-faked.md:47]: Turn on whitespace differences. My editor added a new line. I can expunge if you like.

Comment (thomaslevesque) [reply docs/what-can-be-faked.md:47]: No, don't bother! Thanks

**Comment by thomaslevesque** [comment]
Thanks @blairconrad!

**Comment by blairconrad** [comment]
Thank you, @thomaslevesque.

**Pull request merged**